query-engine/driver-adapters/executor/src/testd.ts

```diff
@@ -217,11 +219,16 @@ function respondOk(requestId: number, payload: unknown) {
 }
 
 async function initQe(url: string, prismaSchema: string, logCallback: qe.QueryLogCallback): Promise<[qe.QueryEngine, ErrorCapturingDriverAdapter]> {
-    const engineType = process.env.EXTERNAL_TEST_EXECUTOR === "Wasm" ? "Wasm" : "Napi";
-    const adapter = await adapterFromEnv(url) as DriverAdapter
-    const errorCapturingAdapter = bindAdapter(adapter)
-    const engineInstance = await qe.initQueryEngine(engineType, errorCapturingAdapter, prismaSchema, logCallback, debug)
-    return [engineInstance, errorCapturingAdapter];
+    if(process.env.EXTERNAL_TEST_EXECUTOR === "Mobile") {
+      const engineInstance = createRNEngineConnector(url, prismaSchema, logCallback)
+      return [engineInstance, {} as any];
```

Comment:
I'm ok with either, this was mostly just test code as the priority was getting the test suite to run. You let me know what you prefer.

Reply:
I would suggest, for this PR, that you make `initQe` return a `Promise<[qe.QueryEngine, ErrorCapturingDriverAdapter | null]`. This will force you to handle the null case in the caller of `initQe`.

That means `state` should equally expect `ErrorCapturingDriverAdapter | null`:
https://github.com/prisma/prisma-engines/blob/2d2b0881f294d475caf2419406c61e1ad1f5d971/query-engine/driver-adapters/executor/src/testd.ts#L80-L84

And this should handle the null case:
https://github.com/prisma/prisma-engines/blob/2d2b0881f294d475caf2419406c61e1ad1f5d971/query-engine/driver-adapters/executor/src/testd.ts#L126

Ezpz nowadays, you can use the `?` operator.

Reply:
done